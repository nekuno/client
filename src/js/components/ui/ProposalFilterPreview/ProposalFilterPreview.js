import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './ProposalFilterPreview.scss';
import RoundedIcon from "../RoundedIcon/RoundedIcon";
import translate from "../../../i18n/Translate";
import FilterStore from "../../../stores/FilterStore";
import connectToStores from "../../../utils/connectToStores";

function getState() {
    const filters = FilterStore.filters;

    return {
        filters,
    };
}

@translate('ProposalFilterPreview')
@connectToStores([FilterStore], getState)
export default class ProposalFilterPreview extends Component {

    static propTypes = {
        // Injected by @translate:
        strings        : PropTypes.object,
        proposalFilters: PropTypes.object,
        // Injected by @connectToStores:
        filters        : PropTypes.object,
    };

    renderProposalFilterArray(filter, choices) {
        let filterItem = '';
        Object.keys(filter).map(function(element, index) {
            choices.map(function(choice, i) {
                if (choice.id === filter[element])
                    filterItem += choice.text;
            });
            if (filter.length !== index + 1 && (filter instanceof Array))
                filterItem += ', ';
        });
        return filterItem;
    }

    renderProposalFilterBirthday(filter) {
        const {strings} = this.props;

        return strings.from + ' ' + filter.min + ' ' + strings.to + ' ' + filter.max + ' ' + strings.years;
    }

    renderProposalFilterLocation(filter) {
        const {strings} = this.props;

        return filter.location.address + ' ' + strings.withinRadioOf + ' ' + filter.distance + ' km';
    }

    renderProposalFilter(item, filter) {
        const {filters} = this.props;

        switch (item) {
            case 'alcohol':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'birthday':
                return this.renderProposalFilterBirthday(filter);
                break;
            case 'civilStatus':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'complexion':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'descriptiveGender':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'ethnicGroup':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'eyeColor':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'hairColor':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'interfaceLanguage':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            case 'location':
                return this.renderProposalFilterLocation(filter);
                break;
            case 'objective':
                return this.renderProposalFilterArray(filter, filters.userFilters[item].choices);
                break;
            default:
                break;
        }
    }

    render() {
        const {strings, proposalFilters, filters} = this.props;
        return (
            <div className={styles.availabilityPreview}>
                <div className={styles.roundedIconWrapper}>
                    <RoundedIcon
                        icon={'eye-off'}
                        size={'small'}
                        color={'#2B3857'}
                        background={'#E9E9E9'}/>
                </div>
                <div className={styles.textWrapper}>
                    <div className={[styles.title,styles.small]}>{strings.filterText}</div>
                </div>
                {Object.keys(proposalFilters).map((item, index) =>
                    <div key={index} className={styles.filterWrapper}>
                        <div className={styles.small}>
                            <strong>{filters.userFilters[item].label}</strong>
                        </div>
                        <div className={styles.small}>{this.renderProposalFilter(item, proposalFilters[item])}</div>
                    </div>
                )}
            </div>
        );
    }
}

ProposalFilterPreview.defaultProps = {
    strings: {
        filterText     : 'Filters to your proposal target',
        from           : 'From',
        to             : 'to',
        years          : 'years',
        withinRadioOf  : 'within radio of'
    }
};