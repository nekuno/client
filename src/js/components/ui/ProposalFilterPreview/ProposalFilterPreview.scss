@import '../../../../scss/_partials/colors';

.availability-preview {
  padding: 16px;
  background: $background-darken;
}

.rounded-icon-wrapper {
  text-align: center;
  display: inline-block;
  margin-right: 4%;
  width: 8%;
  vertical-align: middle;
}

.text-wrapper {
  display: inline-block;
  text-align: left;
  vertical-align: middle;
  width: 88%;
}

.title {
  font-weight: bold;
  margin-bottom: 8px;
  text-align: left;
  color: $text;
}

.small {
  font-size: 14px;
}

.filter-wrapper {
  margin: 16px 0;
}