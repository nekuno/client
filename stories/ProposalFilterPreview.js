import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links'
import ProposalFilterPreview from '../src/js/components/ui/ProposalFilterPreview/ProposalFilterPreview.js';

storiesOf('ProposalFilterPreview', module)
    .add('with text', () => (
        <ProposalFilterPreview onClickHandler={action('clicked')}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam et elit ante.
        </ProposalFilterPreview>
    ))
    .add('with HTML', () => (
        <ProposalFilterPreview onClickHandler={action('clicked')}>
            <div>Lorem ipsum</div>
            <br/>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam et elit ante.</div>
        </ProposalFilterPreview>
    ));