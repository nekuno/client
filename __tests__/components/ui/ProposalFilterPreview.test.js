import React from 'react';
import ProposalFilterPreview from '../../../src/js/components/ui/ProposalFilterPreview/ProposalFilterPreview.js';
import styles from '../../../src/js/components/ui/ProposalFilterPreview/ProposalFilterPreview.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test ProposalFilterPreview component', () => {
    const mockFn = jest.fn();
    const availabilitypreview = shallow(<ProposalFilterPreview onClickHandler={mockFn}>Lorem ipsum</ProposalFilterPreview>);

    it('should be defined', () => {
        expect(availabilitypreview).toBeDefined();
    });
    it('should render correctly', () => {
        expect(availabilitypreview).toMatchSnapshot();
    });
});
