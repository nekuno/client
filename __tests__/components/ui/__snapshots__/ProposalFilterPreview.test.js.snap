// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProposalFilterPreview component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    onClickHandler={[MockFunction]}
    strings={
      Object {
        "filterText": "Filters to your proposal target",
        "from": "From",
        "to": "to",
        "withinRadioOf": "within radio of",
        "years": "years",
      }
    }
  >
    Lorem ipsum
  </TranslationComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Lorem ipsum",
      "locale": undefined,
      "onClickHandler": [MockFunction],
      "strings": Object {
        "filterText": "Filters to your proposal target",
        "from": "From",
        "to": "to",
        "withinRadioOf": "within radio of",
        "years": "years",
      },
    },
    "ref": null,
    "rendered": "Lorem ipsum",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Lorem ipsum",
        "locale": undefined,
        "onClickHandler": [MockFunction],
        "strings": Object {
          "filterText": "Filters to your proposal target",
          "from": "From",
          "to": "to",
          "withinRadioOf": "within radio of",
          "years": "years",
        },
      },
      "ref": null,
      "rendered": "Lorem ipsum",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
